### DISCUSSION TITLE
Charge time estimation

### DISCUSSION THREAD

**Discussion by djpalmis** [teslamate-org/teslamate#2750] [Ideas]
It would be nice to see the estimated time on which car will be charged.
Every time I have to compute on my mind when car will be ready to be used, it is an easy calc for a software..

I.e.

Status | charging for 28 min
Remaining Time | 8 h, 34 min
**Charged at (est.) | 08:12 am (tomorrow)**
Range (rated) | 266.85 km
Range (est.) | 273.23 km
Charged | 0.52 kWh
Charger Power | 2 kW
Charge Limit | 80%
State of Charge | 55% (56%)
Outside Temperature | 27.5 °C
Inside Temperature | 28.4 °C
Mileage | 20000 km
Version | 2022.16.3

**Comment by djpalmis** [comment]
gentle reminder

**Reply by djpalmis** [reply]
Double-gentle reminder

**Comment by djpalmis** [comment]
Bump due to new Administrators.

Anyway thank you Adrian Kumpf (I don't tag him to not bother him)!

**Comment by djpalmis** [comment]
Lol: bard is able to write in Elixir the code I need 🤣
https://g.co/bard/share/2afd0473a1de
Shall I request to pull this code?! 🤣

**Reply by cwanja** [reply]
Push it to a local repo and test it 👍🏼

**Comment by Dulanic** [comment]
This is available in the API, but we need a PR to add it. I don't believe it is stored right now.  Full charge is a misnomer as it is really minutes to charge limit.  You're welcome to make this a feature request issue but no idea when it might be gotten to.

 It is in the API like so:
`"minutes_to_full_charge": 35,`

**Reply by djpalmis** [reply]
I could have sworn that time_to_full_charge was the data I'm looking for.

**Comment by djpalmis** [comment]
I'm not dreaming.. thank you by heart Mr. @NirKli 
Please let me offer a dinner if you ever come to Italy! 
![Screenshot_20240202_211831_Samsung Internet](https://github.com/teslamate-org/teslamate/assets/402122/2a9b5ed9-d290-482b-b7ea-2e4b0be07368)

**Reply by DFOUD** [reply]
This seems to be in the wrong time zone in China, it's now February 4, but the estimated time is on February 3

**Reply by NirKli** [reply]
@DFOUD 
Have you changed the default TZ env var in docker compose file?
https://docs.teslamate.org/docs/configuration/environment_variables

**Comment by prussell69** [comment]
Is it possible to get this info on a grafana screen? It would be nice to display on the charging displays.

**Reply by DrMichael** [reply]
AFAICS, we don't have `time_to_full_charge` in the database...
